### ISSUE TITLE
Error on effect size on Kruskall-Wallis test 

### ISSUE THREAD

**Issue by phisanti** [easystats/effectsize#576]
**Describe the bug**
I got the following error when running effect size on a kruskal.wallis test:

```
Error: Unable to retrieve data from htest object.
  Try using 'rank_epsilon_squared()' directly.
In addition: Warning message:
In grepl("Kruskal-Wallis", x$method, fixed = TRUE) && startsWith(data_name,  :
  'length(x) = 2 > 1' in coercion to 'logical(1)'
```

**To Reproduce**

```R

library(effectsize)
library(data.table)

d <- data.table(iris)

# Error
z <- kruskal.test(Sepal.Length ~ Species, data = d)
effectsize(z)

# No error here
z <- aov(Sepal.Length ~ Species, data = d)
effectsize(z)

```
[conciser using `reprex::reprex()`]

**Expected behavior**
The expected behaviour should be the output as described for `AOV` for example. The package `rstatix` has the function kruskal_effsize, so I believe it should be possible.

**Specifiations (please complete the following information):**
 - `R` Version [R version 4.2.2]
 - `effectsize` Version [0.8.3]

**Comment by strengejacke** [comment]
The error is intended, as `effectsize()` is not able to recover the data from `kruskal.test()`. However, the additional warning was a bug in _insight_, which is fixed.

To solve your problem, use `rank_epsilon_squared()` as suggested by the warning message:

``` r
library(effectsize)
library(data.table)

d <- data.table(iris)

z <- kruskal.test(Sepal.Length ~ Species, data = d)
effectsize(z)
#> Error: Unable to retrieve data from htest object.
#>   Try using 'rank_epsilon_squared()' directly.

rank_epsilon_squared(Sepal.Length ~ Species, data = d)
#> Epsilon2 (rank) |       95% CI
#> ------------------------------
#> 0.65            | [0.58, 1.00]
#> 
#> - One-sided CIs: upper bound fixed at [1.00].

z <- aov(Sepal.Length ~ Species, data = d)
effectsize(z)
#> For one-way between subjects designs, partial eta squared is equivalent
#>   to eta squared. Returning eta squared.
#> # Effect Size for ANOVA
#> 
#> Parameter | Eta2 |       95% CI
#> -------------------------------
#> Species   | 0.62 | [0.54, 1.00]
#> 
#> - One-sided CIs: upper bound fixed at [1.00].
```

<sup>Created on 2023-03-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by mattansb** [comment]
Or you can use the vector inputs to `kruskal.test()`

``` r
z <- kruskal.test(iris$Sepal.Length, iris$Species)

effectsize::effectsize(z)
#> Epsilon2 (rank) |       95% CI
#> ------------------------------
#> 0.65            | [0.60, 1.00]
#> 
#> - One-sided CIs: upper bound fixed at [1.00].
```

Which gives the same as:

``` r
effectsize::rank_epsilon_squared(iris$Sepal.Length, iris$Species)
#> Epsilon2 (rank) |       95% CI
#> ------------------------------
#> 0.65            | [0.57, 1.00]
#> 
#> - One-sided CIs: upper bound fixed at [1.00].
```

<sup>Created on 2023-03-13 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
